python/python/lance/dataset.py

```diff
@@ -1988,6 +1989,17 @@ def write_dataset(
         a custom class that defines hooks to be called when each fragment is
         starting to write and finishing writing.
     """
+    try:
+        # Huggingface datasets
+        import datasets
+
+        if isinstance(data_obj, datasets.Dataset):
+            if schema is None:
+                schema = data_obj.features.arrow_schema
+            data_obj = data_obj.data.to_batches()
+    except ImportError:
+        pass
+
```

Comment:
Please make this import lazy. Right now it will import huggingface anytime you call this function. It should only import if the object is from the huggingface datasets module.

For how we've done this for Polars and other modules, see: https://github.com/lancedb/lance/blob/main/python/python/lance/dependencies.py

```suggestion
    if _check_for_huggingface(data_obj):
        # Huggingface datasets
        import datasets

        if isinstance(data_obj, datasets.Dataset):
            if schema is None:
                schema = data_obj.features.arrow_schema
            data_obj = data_obj.data.to_batches()

```

Reply:
Done